types/chrome/index.d.ts

```diff
@@ -311,6 +311,14 @@ declare namespace chrome.action {
      */
     export function setBadgeText(details: BadgeTextDetails, callback?: () => void): void;
 
+    /**
+     * Since Chrome 88.
+     * Sets the icon for the action. The icon can be specified either as the path to an image file or as the pixel data from a canvas element,
+     * or as dictionary of either one of those. Either the path or the imageData property must be specified.
+     * @return The `setIcon` method provides its result via callback or returned as a `Promise` (MV3 only). Since Chrome 96.
+     */
+    export function setIcon(details: TabIconDetails): Promise<void>;
```

Comment:
In the non-promisified API below, `callback` should not be optional. When it's omitted, the function returns a Promise.

Reply:
See https://github.com/fregante/DefinitelyTyped/pull/1

Reply:
> +1, that should be preferred. In non-promise versions, callback should not be optional. Can you update line 329 to make the callback not optional?

I've updated it.